diff --git a/node_modules/@shopify/flash-list/dist/recyclerview/RecyclerView.js b/node_modules/@shopify/flash-list/dist/recyclerview/RecyclerView.js
--- a/node_modules/@shopify/flash-list/dist/recyclerview/RecyclerView.js
+++ b/node_modules/@shopify/flash-list/dist/recyclerview/RecyclerView.js
@@ -308,16 +308,15 @@
         }
         return onScrollHandler;
     }, [onScrollHandler, scrollY, stickyHeaders, stickyHeaderUseNativeDriver]);
-    const shouldMaintainVisibleContentPosition = recyclerViewManager.shouldMaintainVisibleContentPosition();
     const maintainVisibleContentPositionInternal = useMemo(() => {
-        if (shouldMaintainVisibleContentPosition) {
+        if (maintainVisibleContentPosition != null) {
             return {
                 ...maintainVisibleContentPosition,
                 minIndexForVisible: 0,
             };
         }
         return undefined;
-    }, [maintainVisibleContentPosition, shouldMaintainVisibleContentPosition]);
+    }, [maintainVisibleContentPosition]);
     const shouldRenderFromBottom = recyclerViewManager.getDataLength() > 0 &&
         ((_d = maintainVisibleContentPosition === null || maintainVisibleContentPosition === void 0 ? void 0 : maintainVisibleContentPosition.startRenderingFromBottom) !== null && _d !== void 0 ? _d : false);
     // Create view for measuring bounded size
@@ -329,11 +328,11 @@
             }, ref: firstChildViewRef }));
     }, [horizontal, stickyHeaderOffset]);
     const scrollAnchor = useMemo(() => {
-        if (shouldMaintainVisibleContentPosition) {
+        if (maintainVisibleContentPosition != null) {
             return (React.createElement(ScrollAnchor, { horizontal: Boolean(horizontal), scrollAnchorRef: scrollAnchorRef }));
         }
         return null;
-    }, [horizontal, shouldMaintainVisibleContentPosition]);
+    }, [horizontal, maintainVisibleContentPosition]);
     // console.log("render", recyclerViewManager.getRenderStack());
     // Render the main RecyclerView structure
     return (React.createElement(RecyclerViewContextProvider, { value: recyclerViewContext },
diff --git a/node_modules/@shopify/flash-list/src/recyclerview/RecyclerView.tsx b/node_modules/@shopify/flash-list/src/recyclerview/RecyclerView.tsx
--- a/node_modules/@shopify/flash-list/src/recyclerview/RecyclerView.tsx
+++ b/node_modules/@shopify/flash-list/src/recyclerview/RecyclerView.tsx
@@ -480,18 +480,15 @@
     return onScrollHandler;
   }, [onScrollHandler, scrollY, stickyHeaders, stickyHeaderUseNativeDriver]);
 
-  const shouldMaintainVisibleContentPosition =
-    recyclerViewManager.shouldMaintainVisibleContentPosition();
-
   const maintainVisibleContentPositionInternal = useMemo(() => {
-    if (shouldMaintainVisibleContentPosition) {
+    if (maintainVisibleContentPosition != null) {
       return {
         ...maintainVisibleContentPosition,
         minIndexForVisible: 0,
       };
     }
     return undefined;
-  }, [maintainVisibleContentPosition, shouldMaintainVisibleContentPosition]);
+  }, [maintainVisibleContentPosition]);
 
   const shouldRenderFromBottom =
     recyclerViewManager.getDataLength() > 0 &&
@@ -512,7 +509,7 @@
   }, [horizontal, stickyHeaderOffset]);
 
   const scrollAnchor = useMemo(() => {
-    if (shouldMaintainVisibleContentPosition) {
+    if (maintainVisibleContentPosition != null) {
       return (
         <ScrollAnchor
           horizontal={Boolean(horizontal)}
@@ -521,7 +518,7 @@
       );
     }
     return null;
-  }, [horizontal, shouldMaintainVisibleContentPosition]);
+  }, [horizontal, maintainVisibleContentPosition]);
 
   // console.log("render", recyclerViewManager.getRenderStack());
 
